### PR TITLE
 Cleanup and enhance DateTimeFormatInfo tests

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs [hunk 1]
```diff
@@ -2,36 +2,93 @@
...
+            var value = new string[] { "1", "2", "3", "4", "5", "6", "7" };
+            var format = new DateTimeFormatInfo
+            {
+                AbbreviatedDayNames = value
+            };
+            value[0] = null;
+
+            var dateTime = new DateTime(2014, 5, 28);
+            Assert.Throws<NullReferenceException>(() => dateTime.ToString("ddd MMM yy", format));
+        }
```
Comment (stephentoub) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs]: I suggest we don't need a test validating that we null ref here and in similar places.  The setter does its best to validate that the input is valid, but cloning would be unnecessarily expensive, and if you explicitly then break it by setting an element of the array to null, that's on you.

File: src/System.Globalization/tests/System/Globalization/TextInfoTests.cs [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void ToLower_Netcore(string name, string str, string expected)
+        {
+            TestToLower(name, str, expected);
+        }
+
+        [Theory]
+        [MemberData(nameof(ToLower_TestData_net46))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Netcoreapp | TargetFrameworkMonikers.Uap)]
+        public void ToLower_net46(string name, string str, string expected)
```
Comment (stephentoub) [inline src/System.Globalization/tests/System/Globalization/TextInfoTests.cs]: I believe we decided to stop adding netfx-specific tests.
cc: @danmosemsft

**Comment by stephentoub** [comment]
cc: @tarekgh, @krwq, @GrabYourPitchforks

File: src/System.Globalization/tests/System/Globalization/TextInfoTests.cs (hunk 2 above)
Comment (danmosemsft) [reply src/System.Globalization/tests/System/Globalization/TextInfoTests.cs]: Correct, we want to remove those where an assembly is not shipped for .NET Framework. This assembly is not.
I should have tagged you @hughbe on the discussion/reasoning here: https://github.com/dotnet/corefx/issues/35088

**Comment by tarekgh** [comment]
@hughbe coud you please address @stephentoub comment https://github.com/dotnet/corefx/pull/35944#pullrequestreview-212815649? Thanks.

**Comment by tarekgh** [comment]
@hughbe did you have a chance to address the comment on this PR? thanks.

**Comment by tarekgh** [comment]
@hughbe never mind as I removed the test method as suggested by the comment.

**Comment by tarekgh** [comment]
Looks the failures are unrelated. 

https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F35944~2Fmerge/test~2Ffunctional~2Fcli~2F/20190318.12/workItem/System.Security.Cryptography.X509Certificates.Tests/analysis/xunit/System.Security.Cryptography.X509Certificates.Tests.ChainTests~2FBuildInvalidSignatureTwice

and tracked by old issue https://github.com/dotnet/corefx/issues/30162

CC @bartonjs

**Comment by bartonjs** [comment]
> Looks the failures are unrelated. ... and tracked by old issue #30162

The failures don't look like that issue (they don't run under the remote test executor)

**Comment by tarekgh** [comment]
I have logged a new issue https://github.com/dotnet/corefx/issues/36124 for the crypto failures.

**Pull request merged**